hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -454,13 +454,13 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		ObservedGeneration: nodePool.Generation,
 	})
 
-	// Validate arch platform support
-	if (nodePool.Spec.Arch != "") && (nodePool.Spec.Platform.Type != hyperv1.AWSPlatform) {
+	// Validate modifying CPU arch support for platform
+	if (nodePool.Spec.Arch != "amd64") && (nodePool.Spec.Platform.Type != hyperv1.AWSPlatform) {
 		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
 			Type:               hyperv1.NodePoolValidArchPlatform,
 			Status:             corev1.ConditionFalse,
-			Reason:             hyperv1.NodePoolValidationFailedReason,
-			Message:            fmt.Sprintf("Arch flag is not supported for platform: %s", nodePool.Spec.Platform.Type),
+			Reason:             hyperv1.NodePoolInvalidArchPlatform,
+			Message:            fmt.Sprintf("modifying CPU arch from 'amd64' not supported for platform: %s", nodePool.Spec.Platform.Type),
```

Comment:
I would say: "modifying CPU arch from 'amd64' to unsupported for platform: %s"

Reply:
Hm, but the issue is a particular cloud platform doesn't support modifying the CPU arch away from amd64. Modifying it as suggested, sounds like we are changing the CPU arch to a cloud platform.

Reply:
I was using the wording because we default to amd64 [here](https://github.com/openshift/hypershift/blob/ad1704556cccc267611db9956b04e1431f0c3d68/api/v1beta1/nodepool_types.go#L168).

With that in mind, maybe `CPU arch %s is not supported for platform %s, use 'amd64' instead`?

Reply:
yeah sounds better to me.

Reply:
Thanks Ian and Mulham! This has been updated. Resolving conversation thread.